text/0745-user-provided-cdk-init-templates.md

```diff
@@ -0,0 +1,457 @@
+* **name:** This refers to a variable input which will be the name of your CDK project’s target directory.
+* .**PascalCased:** This is a transformation function that converts the string into PascalCase
+
+
+Ticking the box below indicates that the public API of this RFC has been signed-off by the API bar raiser (the `status/api-approved` label was applied to the RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+
+
+## Public FAQ
+
+### What are we launching today?
+
+A new feature in the AWS CDK CLI that enables users to use custom templates for project initialization from local files, Git repositories, or NPM packages.
+
+### Why should I use this feature?
+
+* Be able to use custom components and configurations by default
+* Share templates for specific use cases through repository or NPM package
+
+
+
+## Internal FAQ
+
+### Why are we doing this?
+
+Users are currently limited to three built-in templates. Organizations need custom templates to:
+
+* Quickly create CDK projects with specific components and requirements and reduce setup time
+
+### Why should we *not* do this?
+
+* Increased CLI complexity
+
+### What is the technical solution (design) of this feature?
+
+Extend `cdk init` with:
+
+1. Custom template source options (local/Git Repo/NPM) with all CDK supported languages
+2. Simple CLI syntax for users to run `cdk init` commands with custom templates
+3. Custom template validation to ensure a valid CDK project can be initialized
+4. Initialize project using custom template
+
+### Is this a breaking change?
+
+No, the new feature does not affect existing functionality for creating CDK projects using built-in template options.
+
+### What alternative solutions did you consider?
+
+1. Alternatives for supporting custom templates for `cdk init`
+    1. Expanded template registry that users can choose from beyond the three built-in templates
+2. Alternatives for Consistent Naming of Template Components
+    1. Current CDK Approach: %name.PascalCased%
+        1. Pros: 
+            1. Already works for built-in templates
+                1. Proven in production with minimal maintenance 
+            2. Independent of template users
+                1. Uses the `cdk init` context (like project folder name), so no user input required.
+            3. Easy format for template authors to use
+                1. Simple substitution syntax with no setup or tooling required.
+            4. No external dependencies
+                1. Entire process is handled within CDK so reduced risk of integration issues
+        2. Cons:
+            1. Limited to filename and file content replacement
+                1. Does not provide interactive prompts or advanced templating logic like Yeoman
+            2. Only helps template authors, not users
+                1. Does not provide a better UX for users who want to customize scaffolding more granularly
```

Comment:
Not sure I follow, how would users benefit from templating, if not indirectly from the flexibility of the template?